File: apps/web/src/types/chapter.ts
export interface Chapter {
  id: string;
  project_id: string;
  title: string;
  sort_order: number;
  current_text: string;
  status: string;
  created_at: string;
  updated_at: string;
  deleted_at: string | null;
}

export interface ChapterListSchema {
  id: string;
  project_id: string;
  title: string;
  sort_order: number;
  status: string;
  created_at: string;
  updated_at: string;
  deleted_at: string | null;
}

export interface ChapterVersion {
  id: string;
  chapter_id: string;
  version_number: number;
  source: string;
  text: string;
  note: string;
  generation_candidate_id: string | null;
  created_at: string;
}

export interface ChapterCreate {
  title?: string;
  sort_order?: number;
  current_text?: string;
}

export interface ChapterUpdate {
  title?: string;
  current_text?: string;
  expected_updated_at: string;
}

export interface ChapterReorderItem {
  id: string;
  sort_order: number;
}

export interface VersionCreate {
  text: string;
  note?: string;
  source?: string;
}

File: apps/api/app/schemas/chapter.py
from datetime import datetime
from pydantic import BaseModel, Field


class ChapterCreate(BaseModel):
    title: str = Field(default="", max_length=200)
    sort_order: int = Field(default=0, ge=0)
    current_text: str = ""


class ChapterUpdate(BaseModel):
    title: str | None = Field(default=None, max_length=200)
    current_text: str | None = None
    expected_updated_at: datetime


class ChapterReorderItem(BaseModel):
    id: str
    sort_order: int


class ChapterReorder(BaseModel):
    items: list[ChapterReorderItem]


class ChapterVersionSchema(BaseModel):
    model_config = {"from_attributes": True}

    id: str
    chapter_id: str
    version_number: int
    source: str
    text: str
    note: str
    generation_candidate_id: str | None
    created_at: datetime


class ChapterSchema(BaseModel):
    model_config = {"from_attributes": True}

    id: str
    project_id: str
    title: str
    sort_order: int
    current_text: str
    status: str
    created_at: datetime
    updated_at: datetime
    deleted_at: datetime | None = None


class ChapterListSchema(BaseModel):
    model_config = {"from_attributes": True}

    id: str
    project_id: str
    title: str
    sort_order: int
    status: str
    created_at: datetime
    updated_at: datetime
    deleted_at: datetime | None = None
